### PR TITLE
[11.0] backport sale_timesheet_rounded from 12.0

### PR DESCRIPTION
**Pull request by grindtildeath**
Backport of #257

### CODE REVIEW

File: project_task_stage_allow_timesheet/models/account_analytic_line.py [hunk 1]
```diff
@@ -15,6 +16,11 @@ class AccountAnalyticLine(models.Model):
 
     @api.constrains('task_id')
     def _check_task_allow_timesheet(self):
+        if (
+            config['test_enable'] and
+            not self.env.context.get('test_task_stage_allow_timesheet')
+        ):
+            return
```
Comment (i-vyshnevska) [inline project_task_stage_allow_timesheet/models/account_analytic_line.py:23]: this is weird tests shouldn't imply on  logic in method

File: project_task_stage_allow_timesheet/tests/test_project_task_stage_allow_timesheet.py [hunk 2]
```diff
@@ -10,6 +10,10 @@ class TestProjectTaskStageAllowTimesheet(SavepointCase):
     @classmethod
     def setUpClass(cls):
         super(TestProjectTaskStageAllowTimesheet, cls).setUpClass()
+        cls.env = cls.env(context=dict(
+            cls.env.context, tracking_disable=True,
+            test_task_stage_allow_timesheet=True
+        ))
```
Comment (i-vyshnevska) [inline project_task_stage_allow_timesheet/tests/test_project_task_stage_allow_timesheet.py:16]: maybe additional flags needs to force tests

File: sale_timesheet_rounded/models/account_analytic_line.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    @api.onchange('unit_amount')
+    def _onchange_unit_amount(self):
+        self.unit_amount_rounded = self.with_context(
+            force_compute=1)._calc_unit_amount_rounded()
+
+    def _calc_unit_amount_rounded(self):
+        self.ensure_one()
+        force_compute = self.env.context.get('force_compute')
+        # TODO: do we still need this?
```
Comment (i-vyshnevska) [inline sale_timesheet_rounded/models/account_analytic_line.py:50]: or more explicit comment or remove it

**Review by leemannd — APPROVED** [review]
Tested functionally

**Comment by leemannd** [comment]
@simahawk @alexey-pelykh If you have time to take a look at it. It's worth merging

**Review by alexey-pelykh — COMMENTED** [review]
Why commit history is so odd, if sale_timesheet_rounded was added via 1 commit in 12?

**Comment by leemannd** [comment]
@alexey-pelykh You're right about the commit history.
@grindtildeath do you think you can update this?

**Comment by grindtildeath** [comment]
@alexey-pelykh @leemannd Actually, the module wasn't added via 1 commit in 12... At the beginning were the first 6 commits from @simahawk that you see here. Then I fixed the issues with all of the commits you see here on 11.0, which have been forward ported into #257 (https://github.com/simahawk/timesheet/pulls?q=is%3Apr+is%3Aclosed) before being squashed (without taking care of the author) into the same one, after the last commit added by @simahawk here...

So this commit history was actually "the right one", but now I squashed by authors and kept important commits messages so that we can merge...

**Comment by leemannd** [comment]
@grindtildeath Thanks! :+1:

**Pull request closed without merging**